npm/pod.go

```diff
@@ -0,0 +1,140 @@
+package npm
+
+import (
+	"strings"
+
+	"github.com/Azure/azure-container-networking/log"
+	"github.com/Azure/azure-container-networking/npm/util"
+
+	corev1 "k8s.io/api/core/v1"
+)
+
+func isValidPod(podObj *corev1.Pod) bool {
+	return podObj.Status.Phase != "Failed" &&
+		podObj.Status.Phase != "Succeeded" &&
+		podObj.Status.Phase != "Unknown" &&
+		len(podObj.Status.PodIP) > 0
+}
+
+func isSystemPod(podObj *corev1.Pod) bool {
+	return podObj.ObjectMeta.Namespace == util.KubeSystemFlag
+}
+
+// AddPod handles add pod.
+func (npMgr *NetworkPolicyManager) AddPod(podObj *corev1.Pod) error {
+	npMgr.Lock()
+	defer npMgr.Unlock()
+
+	if !isValidPod(podObj) {
+		return nil
+	}
+
+	podNs := podObj.ObjectMeta.Namespace
+	podName := podObj.ObjectMeta.Name
+	podNodeName := podObj.Spec.NodeName
+	podLabels := podObj.ObjectMeta.Labels
+	podIP := podObj.Status.PodIP
+	log.Printf("POD CREATED: %s/%s/%s%+v%s\n", podNs, podName, podNodeName, podLabels, podIP)
+
+	// Add the pod to ipset
+	ipsMgr := npMgr.nsMap[util.KubeAllNamespacesFlag].ipsMgr
+	// Add the pod to its namespace's ipset.
+	log.Printf("Adding pod %s to ipset %s\n", podIP, podNs)
+	if err := ipsMgr.AddToSet(podNs, podIP); err != nil {
+		log.Printf("Error adding pod to namespace ipset.\n")
```

Comment:
In case of error, why are we still going ahead with the rest of the code?

Reply:
addressed